### PR TITLE
developer-workflow: finalize Phase D test-coverage-expert conditional trigger

### LINKED ISSUE #152: developer-workflow: test-coverage-expert conditional trigger в finalize Phase D
Parent: #151

## Context

`plugins/developer-workflow/skills/finalize/SKILL.md` Phase D запускает conditional expert reviews (security-expert, architecture-expert, perf, ux). Сейчас нет эксперта, проверяющего coverage vs TestPlan.

## Problem

Тесты внутри Implement могут оказаться неполными:
- Engineer-агент пропустил edge cases, заявленные в TestPlan.
- TestPlan содержит TC определённых типов, код-тесты их не реализовали.
- В diff появился новый public API без соответствующего test-файла.

Second-pass для этого отсутствует.

## Proposal

Добавить conditional trigger `test-coverage-expert` в Finalize Phase D.

### Условия срабатывания (любое)

1. Diff добавил новый public API (class / interface / public function / module entry) без соответствующего test-файла.
2. TestPlan в `docs/testplans/<slug>-test-plan.md` содержит TC, не реализованные в test-коде для этого slug.
3. Diff затрагивает `data layer / repository / service` без новых test-файлов.
4. `--coverage-audit` явный override.

### Skip conditions (любое)

1. Trivial diff (single-file, <50 LOC, нет нового public API, refactor-only).
2. `--skip-coverage-audit` явный override.
3. В проекте нет test-infrastructure (short-circuit, предложение отдельного issue).

### Что делает expert

1. Читает TP, AC из spec/plan, diff.
2. Перечисляет test-файлы, добавленные в diff.
3. Кросс-референс: для каждого TC из TP — есть ли test, покрывающий его (по имени + содержимому, интерпретируется engineer-агентом, не regex)?
4. Для каждого нового public API — есть ли test?
5. Генерирует `swarm-report/<slug>-coverage-audit.md` с gap-list'ом.
6. Если gaps → делегирует engineer-агенту добор. Engineer пишет недостающие тесты, `/check` gate.
7. Если gaps нет → Finalize Phase D продолжается.

### Реализация эксперта

Не новый agent-файл — переиспользуем существующих engineer-агентов (kotlin-engineer, swift-engineer, compose-developer, swiftui-developer). Skill вызывает Task tool с платформенно-специфичным промптом «audit test coverage».

Min-bar (issue #148): не плодим новый agent-role без необходимости.

## Implementation

- `plugins/developer-workflow/skills/finalize/SKILL.md` Phase D — добавить строку trigger'а с условиями активации и skip.
- `plugins/developer-workflow/docs/ORCHESTRATION.md` — обновить, если там сводная таблица triggers.
- Schema артефакта `<slug>-coverage-audit.md` документировать.
- Override `--skip-coverage-audit` в секции Overrides.

## Acceptance criteria

- [ ] `finalize/SKILL.md` Phase D содержит `test-coverage-expert` trigger с trigger/skip conditions.
- [ ] Trigger документирован в docs/ORCHESTRATION.md при наличии сводной таблицы.
- [ ] Schema `<slug>-coverage-audit.md` задокументирована.
- [ ] Smoke test (positive): feature с новым public class без теста → trigger срабатывает → добавлен тест → `/check` PASS.
- [ ] Smoke test (skip): trivial refactor → trigger молча пропускается.
- [ ] `--skip-coverage-audit` работает, фиксируется в finalize artifact.
- [ ] Нет нового agent-файла — используются existing engineer-agents.

## Non-goals

- Не создаём новый agent.
- Не численный coverage threshold.
- Не переписываем тесты, написанные engineer'ом в Implement — только дополняем.

## Dependencies

- Parent: #151.
- Depends on: child про `type` field в generate-test-plan — нужен для precise TP ↔ test cross-reference.
- Complements: child про `/check` gate (ранняя отсечка, до Finalize).

## Effort / Risk

- Effort: S–M (markdown + prompt skeleton).
- Risk: false-negatives при наивной name-heuristic. Митигация: engineer-агент интерпретирует содержимое, не regex.

### PR DESCRIPTION
**Pull request by kirich1409**
## Summary

Closes #152.

- `finalize/SKILL.md` Phase D gains a **`test-coverage-expert` (conditional)** section: trigger-when-any list (new public API without matching test, TP Test Cases not implemented in tests, data layer touched without new tests, `--coverage-audit` force-on), skip-when-any list (trivial diff, `--skip-coverage-audit`, no test infrastructure with follow-up issue), and a fully documented schema for `swarm-report/<slug>-coverage-audit.md`.
- The role reuses **existing engineer agents** (`kotlin-engineer`, `swift-engineer`, `compose-developer`, `swiftui-developer`) with a coverage-audit prompt — no new agent file. Honours min-bar item 3 (no duplication, #148 / PR #175).
- Verdict mapping: `PASS` / `GAPS_RESOLVED` / `ESCALATE`. `GAPS_RESOLVED` requires the agent to write missing tests AND `/check` PASS in the same call (author-fixes-tests rule from #157 applies).
- Two new Tolerance flags on `finalize`:
  - `--coverage-audit` — force-on the trigger even when no diff condition matches; useful for explicit pre-release sweeps.
  - `--skip-coverage-audit` — turn off the trigger; recorded verbatim with the user reason in the finalize report's acknowledged-risks section.
- `docs/ORCHESTRATION.md` Phase D trigger table grows a row for `test-coverage-expert`, with a note that the role reuses existing engineer agents and the detailed rules live in `finalize/SKILL.md`.

## Acceptance criteria mapping (from #152)

- [x] `finalize/SKILL.md` Phase D contains a `test-coverage-expert` trigger with trigger / skip conditions.
- [x] Trigger documented in `docs/ORCHESTRATION.md` Phase D trigger table.
- [x] Schema for `swarm-report/<slug>-coverage-audit.md` documented.
- [ ] Smoke test (positive): feature with a new public class without test → trigger fires → engineer writes test → `/check` PASS — verifiable on the next real run.
- [ ] Smoke test (skip): trivial refactor → trigger silently skipped — verifiable on the next real run.
- [x] `--skip-coverage-audit` flag works and is recorded in the finalize report.
- [x] No new agent file — uses existing engineer agents.

## Notes for review

- **Complementary to #154 (PR #182), not a replacement.** `/check` Phase 3.5 is the early gate (caught at Implement time); Phase D `test-coverage-expert` is the late audit (caught at Finalize time, can also cross-reference declared TCs against code, which the early gate does not do). The two together cover both the "engineer wrote a public API and forgot a test" failure mode and the "test plan declared a TC that nobody implemented" failure mode.
- **No new agent role.** Per the parent strategy (#151) and the min-bar checklist (#148), the trigger is implemented as a different prompt to the existing engineer agents — not a new agent file. This keeps the agent inventory honest and avoids overlap with the engineers that already understand the test conventions.

## Out of scope

- Numerical coverage threshold (#152 Non-goals).
- Rewriting tests already produced by the engineer in Implement (#152 Non-goals — the audit only adds, never rewrites).
- A new agent file (#152 Non-goals).

## Dependencies

- Forward link to `docs/TESTING-STRATEGY.md` (#151 / PR #178) and to the author-fixes-tests rule (#157 / PR #179) — both resolve once those PRs merge.
- The cross-reference between TC `Type` and test files relies on the type field (#153 / PR #181).
- Complements `/check` Phase 3.5 (#154 / PR #182) — early vs late audit.

## Test plan

- [x] `bash scripts/validate.sh` — green.
- [ ] Manual: introduce a feature with a new public class lacking a test → run `finalize` → confirm the audit triggers and the engineer writes a test before round-end.
- [ ] Manual: confirm `--skip-coverage-audit` skips the audit and records the reason in the finalize report.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: plugins/developer-workflow/skills/finalize/SKILL.md [hunk 1]
```diff
@@ -160,6 +162,77 @@ Experts produce deeper, higher-risk findings. Apply the same severity × confide
...
+
+1. The diff adds a new public API symbol that has no matching test file.
+2. `docs/testplans/<slug>-test-plan.md` declares Test Cases that have no matching implementation in the test sources for this slug. Cross-reference is by Test Case `Type` (#153) plus name / file mention — interpreted by the engineer agent, not regex.
+3. The diff touches data layer / repository / service / use-case files without introducing or updating tests for them.
+4. The caller passed `--coverage-audit` (force-on).
+
+**Skip when ANY:**
+
+1. The diff is trivial (single file, < 50 LOC, no new public API, refactor-only).
+2. The caller passed `--skip-coverage-audit` (recorded in the finalize report verbatim with the user reason; required for the orchestrator-driven invocation, see [`docs/TESTING-STRATEGY.md`](../../docs/TESTING-STRATEGY.md#skip-rules)).
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/finalize/SKILL.md:179]: Resolved: PR #178 introduced `plugins/developer-workflow/docs/TESTING-STRATEGY.md` on main. Branch rebased onto main; the `#skip-rules` anchor (`## Skip rules` section header) exists in the file. Link target is now valid.
